pkg/queue/prometheus_stats_reporter.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package queue
+
+import (
+	"errors"
+	"net/http"
+	"time"
+
+	"github.com/knative/serving/pkg/autoscaler"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+const (
+	// ReporterReportingPeriod is the interval of time between reporting stats by queue proxy.
+	ReporterReportingPeriod = time.Second
+
+	destinationNsLabel     = "destination_namespace"
+	destinationConfigLabel = "destination_configuration"
+	destinationRevLabel    = "destination_revision"
+	destinationPodLabel    = "destination_pod"
+)
+
+var (
+	metricLabelNames = []string{
+		destinationNsLabel,
+		destinationConfigLabel,
+		destinationRevLabel,
+		destinationPodLabel,
+	}
+
+	// TODO(#2524): make reporting period accurate.
+	operationsPerSecondGV = prometheus.NewGaugeVec(
```

Comment:
This seems repeatable. May be a function `func newGV(n, h string) { return prom...}`